### PR TITLE
Experiment

### PR DESCRIPTION
**Pull request by tiffapedia**
Create an experiment api. Still need to put differentiator in timestamp and cleanup comments in code.

### CODE REVIEW

File: experiment_api/experiment.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+# -----------------------------------------------------------------------------
+#  Execution
+# -----------------------------------------------------------------------------
+
+def main():
+    # inputs
+    veri_unzipped_path = ""
+    num_cams = 5
+    num_cars_per_cam = 3
+    drop_percentage = 30
```
Comment (agude) [inline experiment_api/experiment.py]: Can we make `drop_percentage` a float ranging from [0, 1]? This allows more fine-grained percentages.

File: experiment_api/experiment.py [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+        self.name = img_name
+        self.filepath = img_dir + "/" + img_name
+        self.type = img_type
+        self.__splitter = self.name.split("_")
+        self.car_id = self.__set_car_id()
+        self.camera_id = self.__set_camera_id()
+        self.timestamp = self.__set_timestamp()
+        self.binary = self.__set_binary()
+
+    def get_name(self):
```
Comment (agude) [inline experiment_api/experiment.py]: Python doesn't need getters and setters. We can just use `Image.name`. If in the future we need them we can use `@property`.

File: experiment_api/experiment.py [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+        return self.camera_id
+
+    def __set_timestamp(self):
+        timestamp = datetime.datetime.fromtimestamp(int(self.__splitter[2]))
+        return timestamp
+
+    def get_timestamp(self):
+        return self.timestamp
+
+    def get_timestamp_in_str(self):
```
Comment (agude) [inline experiment_api/experiment.py]: This getter is probably ok, but we could also just calculate that and set it as a value.

File: experiment_api/experiment.py [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+    num_cams = 5
+    num_cars_per_cam = 3
+    drop_percentage = 30
+    seed = 11
+
+    # create the generator
+    exp = ExperimentGenerator(veri_unzipped_path, num_cams, num_cars_per_cam, drop_percentage, seed)
+
+    # generate the experiment
+    """
```
Comment (agude) [inline experiment_api/experiment.py]: Is this part still needed?

File: experiment_api/experiment.py [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+    for i in exp.get_images():
+        print("%s: %s: %s: %s: %s: %s: %s" % (i.name, i.filepath, i.type, i.car_id, i.camera_id, i.get_timestamp_in_str(), i.binary))
+    print(len(exp.get_images()))
+    """
+    #print(exp.get_target_car())
+    print(exp.generate())
+    for camset in exp.generate():
+        for i in camset:
+            print("%s: %s: %s: %s: %s: %s: %s" % (i.name, i.filepath, i.type, i.car_id, i.camera_id, i.get_timestamp_in_str(), i.binary))
```
Comment (agude) [inline experiment_api/experiment.py]: Switch to `.format()` instead of `%`.

File: experiment_api/experiment.py (hunk 2 above)
Comment (tiffapedia) [inline experiment_api/experiment.py]: kay, removed

File: experiment_api/experiment.py (hunk 3 above)
Comment (tiffapedia) [inline experiment_api/experiment.py]: it's already calculated, so let me remove that as well

File: experiment_api/experiment.py (hunk 1 above)
Comment (tiffapedia) [inline experiment_api/experiment.py]: done

File: experiment_api/experiment.py (hunk 5 above)
Comment (tiffapedia) [inline experiment_api/experiment.py]: I'm going to commented this out since it is not necessary

**Pull request merged**